addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/NeatoRobot.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neato.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.TimeZone;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.neato.handler.NeatoHandler;
+import org.openhab.binding.neato.internal.classes.ErrorMessage;
+import org.openhab.binding.neato.internal.classes.NeatoGeneralInfo;
+import org.openhab.binding.neato.internal.classes.NeatoRobotInfo;
+import org.openhab.binding.neato.internal.classes.NeatoState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link NeatoBinding} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+
+public class NeatoRobot {
+
+    private static final Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private String serialNumber;
+    private String secret;
+    private String name;
+
+    private NeatoState state;
+    private NeatoRobotInfo info;
+    private NeatoGeneralInfo generalInfo;
+
+    public NeatoState getState() {
+        return this.state;
+    }
+
+    public NeatoRobotInfo getInfo() {
+        return this.info;
+    }
+
+    public NeatoGeneralInfo getGeneralInfo() {
+        return this.generalInfo;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public NeatoRobot(String _serial, String _secret, String _name) {
+        this.serialNumber = _serial;
+        this.secret = _secret;
+        this.name = _name;
+
+        this.state = null;
+        this.info = null;
+        this.generalInfo = null;
+    }
+
+    public String callNeatoWS(String body) throws Exception {
+        SimpleDateFormat dateFormatGmt = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z", Locale.US);
+        dateFormatGmt.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+        // Time in GMT
+        String dateString = dateFormatGmt.format(new Date());
+
+        Mac sha256_HMAC = Mac.getInstance("HmacSHA256");
```

Comment:
The [code guidelines](http://docs.openhab.org/developers/development/guidelines.html#a-code-style) state that we should follow the [Java naming conventions](http://java.about.com/od/javasyntax/a/nameconventions.htm)

In short:

- **Packages**: Names should be in lowercase
- **Classes/Interfaces**: Names should be in CamelCase
- **Methods/Variables**: Names should be in mixed case (CamelCase starting with lower case)
- **Constants**: Names should be in uppercase